### PR TITLE
fix a bug in the latent heat material & add min and max viscosity

### PR DESCRIPTION
**Pull request by jdannberg**
This pull request fixes a bug in the latent heat material model: there was an input parameter `Viscosity prefactors` that was documented and could be set in the input file, but was never used in the code and hence had no actual effect. With these changes, the parameter now does what the documentation says. In addition, parameters for setting a min and max viscosity is introduced. This brings the material to the state we used in our manuscript about numerical methods in ASPECT. 

I also want to see if this changes any tests, if not, I will add one.

### CODE REVIEW

File: source/material_model/latent_heat.cc [hunk 1]
```diff
@@ -208,6 +208,7 @@ namespace aspect
             // this means, that there are no actual density "jumps", but gradual
             // transition between the materials
             double phase_dependence = 0.0;
+            double viscosity_phase_dependence = 1.0;
```
Comment (gassmoeller) [inline source/material_model/latent_heat.cc]: Please update the comment above for this variable

File: source/material_model/latent_heat.cc [hunk 2]
```diff
@@ -228,6 +229,7 @@ namespace aspect
                                                                  i);
 
                     phase_dependence += phaseFunction * density_jumps[i];
+                    viscosity_phase_dependence *= 1 + phaseFunction * (phase_prefactors[i+1]-1);
```
Comment (gassmoeller) [inline source/material_model/latent_heat.cc]: why do you use `phase_prefactors[i+1]` but `density_jumps[i]`? Maybe there is a reason, but this asymmetry looks strange.

File: source/material_model/latent_heat.cc (hunk 1 above)
Comment (gassmoeller) [inline source/material_model/latent_heat.cc]: Also there is a comment in line 181 saying that this part is only computing density, maybe extend that to mention also the phase dependence of the viscosity

File: source/material_model/latent_heat.cc [hunk 3]
```diff
@@ -449,6 +453,14 @@ namespace aspect
                              "viscosity for each phase. "
                              "List must have one more entry than Phase transition depths. "
                              "Units: non-dimensional.");
+          prm.declare_entry ("Minimum viscosity", "1e19",
+                             Patterns::List (Patterns::Double(0)),
```
Comment (gassmoeller) [inline source/material_model/latent_heat.cc]: Should this really be a list? Or just a double?

File: source/material_model/latent_heat.cc [hunk 4]
```diff
@@ -449,6 +453,14 @@ namespace aspect
...
+                             "Limit for the minimum viscosity in the model. "
+                             "Units: Pa s.");
+          prm.declare_entry ("Maximum viscosity", "1e24",
+                             Patterns::List (Patterns::Double(0)),
```
Comment (gassmoeller) [inline source/material_model/latent_heat.cc]: Same here

File: source/material_model/latent_heat.cc [hunk 5]
```diff
@@ -517,8 +535,19 @@ namespace aspect
...
                   phase_prefactors.size() != transition_pressures.size()+1)
-                AssertThrow(false, ExcMessage("Error: At least one list that gives input parameters for the phase transitions has the wrong size. Currently checking against transition pressures. If phase transitions in terms of depth are desired, check to make sure Define transition by depth instead of pressure = true."));
+                AssertThrow(false, ExcMessage("Error: At least one list that gives input parameters for the phase "
+                                              "transitions has the wrong size. Currently checking against transition pressures. "
+                                              "If phase transitions in terms of depth inputs are desired, check to make sure "
+                                              "'Define transition by depth instead of pressure = true'."));
             }
+
+          // as the phase viscosity prefactors are all applied on top of each other, we have
+          // to scale them here so that they are relative gactors in comparison to the phase above
```
Comment (gassmoeller) [inline source/material_model/latent_heat.cc]: factors

Comment (gassmoeller) [inline source/material_model/latent_heat.cc]: I would rephrase this a bit (it took me some time to understand what you mean). What about: "as the phase viscosity prefactors are all applied multiplicatively on top of each other, we have to scale them here so that they are relative factors in comparison to the product of all phase above"

File: source/material_model/latent_heat.cc (hunk 2 above)
Comment (tjhei) [inline source/material_model/latent_heat.cc]: please write the doubles as ``1.``

File: source/material_model/latent_heat.cc [hunk 6]
```diff
@@ -242,6 +244,7 @@ namespace aspect
                       phase_dependence += phaseFunction * density_jumps[i] * (1.0 - composition[0]);
                     else if (transition_phases[i] == 1) // 2nd compositional field
                       phase_dependence += phaseFunction * density_jumps[i] * composition[0];
+                    viscosity_phase_dependence *= 1 + phaseFunction * (phase_prefactors[i]-1);
```
Comment (tjhei) [inline source/material_model/latent_heat.cc]: same here

**Comment by jdannberg** [comment]
Okay, I think I addressed all of your comments, and I also added a test.

File: tests/latent_heat_viscosity.prm [hunk 7]
```diff
@@ -0,0 +1,122 @@
+# A simple setup for for using the GPlates interface in a 2d shell. See the
```
Comment (tjhei) [inline tests/latent_heat_viscosity.prm]: can you update the comment for this test please?

Comment (jdannberg) [inline tests/latent_heat_viscosity.prm]: Okay, fixed.
